server/middleware/validationMiddleware.ts

```diff
@@ -10,56 +11,61 @@ export type FieldValidationError = {
+      } else {
+        classType = (await conditionService.getAdditionalConditionByCode(req.body.code))?.validatorType || type
+      }
+      // Cater for file uploads on specific forms - in this case to setup the filename in the req.body
+      if (req.file && req.file.fieldname === 'outOfBoundFilename') {
+        req.body = { ...req.body, outOfBoundFilename: req.file.originalname }
+      }
 
-    const errors: ValidationError[] = await validate(validationScope)
+      // Build an object which is used by validators to check things against
+      const validationScope = plainToInstance(
+        classType,
+        { ...req.body, licence, uploadFile: req.file },
+        { excludeExtraneousValues: false }
+      )
 
-    if (errors.length === 0) {
-      req.body = plainToInstance(classType, req.body, { excludeExtraneousValues: true })
-      return next()
-    }
+      const errors: ValidationError[] = await validate(validationScope)
 
-    const buildError = (
-      error: ValidationError,
-      constraints: {
-        [type: string]: string
+      if (errors.length === 0) {
+        req.body = plainToInstance(classType, req.body, { excludeExtraneousValues: true })
+        return next()
       }
-    ): FieldValidationError => ({
-      field: error.property,
-      message: Object.values(constraints)[Object.values(constraints).length - 1],
-    })
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const flattenErrors: any = (errorList: ValidationError[]) => {
-      // Flat pack a list of errors with child errors into a 1-dimensional list of errors.
-      return errorList.flatMap(error => {
-        return error.children.length > 0 ? flattenErrors(error.children) : buildError(error, error.constraints)
+      const buildError = (
+        error: ValidationError,
+        constraints: {
+          [type: string]: string
+        }
+      ): FieldValidationError => ({
+        field: error.property,
+        message: Object.values(constraints)[Object.values(constraints).length - 1],
       })
-    }
 
-    req.flash('validationErrors', JSON.stringify(flattenErrors(errors)))
-    req.flash('formResponses', JSON.stringify(req.body))
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const flattenErrors: any = (errorList: ValidationError[]) => {
+        // Flat pack a list of errors with child errors into a 1-dimensional list of errors.
+        return errorList.flatMap(error => {
+          return error.children.length > 0 ? flattenErrors(error.children) : buildError(error, error.constraints)
+        })
+      }
 
-    return res.redirect('back')
+      req.flash('validationErrors', JSON.stringify(flattenErrors(errors)))
+      req.flash('formResponses', JSON.stringify(req.body))
+
+      return res.redirect('back')
+    } catch (error) {
+      logger.error(error, `Failed to validate licence details for: ${req.params.nomisId}`)
```

Comment:
if we log the error then we'll be producing two error events in app insights as the error handler logs it too

Could wrap the error [using the cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#rethrowing_an_error_with_a_cause)

( wouldn't want to rethrow, just pass to `next`)